core/src/main/java/nl/nn/adapterframework/jms/BytesMessageInputStream.java

```diff
@@ -48,12 +54,17 @@ public int read(byte[] b) throws IOException {
 	}
 
 	@Override
-	public int read(byte[] b, int off, int len) throws IOException {
+	public int read(@Nonnull byte[] b, int off, int len) throws IOException {
 		try {
-			byte[] readbuf = new byte[len];
-			int result = bytesMsg.readBytes(readbuf);
+			// If we try filling the whole array from start, then we do not need to read data into
+			// a temp array and copy it.
+			if (off == 0 && len == b.length) {
+				return read(b);
+			}
```

Comment:
Just to understand your fix: this if-statement at line 61 is an optimization right, not a fix?